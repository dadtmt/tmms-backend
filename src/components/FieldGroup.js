import React, { PropTypes } from 'react'
import { ControlLabel, FormControl, FormGroup } from 'react-bootstrap'

const FieldGroup = ({ id, label, ...props }) =>
    <FormGroup controlId={id}>
      {label && <ControlLabel>{label}</ControlLabel>}
      <FormControl {...props} />
    </FormGroup>

FieldGroup.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string
}

export default FieldGroup
