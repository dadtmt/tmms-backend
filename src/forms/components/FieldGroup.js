import React, { PropTypes } from 'react'
import {
  ControlLabel,
  FormGroup,
  FormControl,
  HelpBlock
} from 'react-bootstrap'
import { mapProps } from 'recompose'
import { mapValidationState } from '../utils'

const FieldGroup = ({
  controlId,
  input,
  label,
  meta: { error },
  placeholder,
  size,
  type,
  validationState
}) => (
  <FormGroup
    controlId={controlId}
    bsSize={size ? size : null}
    validationState={validationState ? validationState : null}
  >
    {label && (<ControlLabel>{label}</ControlLabel>)}
    <FormControl {...input} type={type} placeholder={placeholder} />
    {validationState && (<FormControl.Feedback />)}
    <HelpBlock>{error}</HelpBlock>
  </FormGroup>
)

FieldGroup.propTypes = {
  controlId: PropTypes.string.isRequired,
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  meta: PropTypes.object.isRequired,
  placeholder: PropTypes.string,
  size: PropTypes.string,
  type: PropTypes.string.isRequired,
  validationState: PropTypes.string
}

export const ValidableFieldGroup = mapProps(mapValidationState)(FieldGroup)

export default FieldGroup
