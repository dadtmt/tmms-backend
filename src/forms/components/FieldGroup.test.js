import React from 'react'
import { Field, reducer as formReducer, reduxForm } from 'redux-form'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import renderer from 'react-test-renderer'

import FieldGroup, { ValidableFieldGroup } from './FieldGroup'

describe('FieldGroup', () => {
  it('should render correctly with required props', () => {
    const input = {},
      meta = {}
    const output = renderer.create(
      <FieldGroup
        controlId='fieldGroupControlId'
        input={input}
        meta={meta}
        type='text'
      />
    )
    expect(output.toJSON()).toMatchSnapshot()
  })
  it('should render correctly password input', () => {
    const store = createStore(combineReducers({ form: formReducer }))
    const Form = () => <Field
      name='password'
      controlId='password'
      component={ValidableFieldGroup}
      type='password'
      placeholder='password'
      size='large'
    />
    const ReduxForm = reduxForm({ form: 'password' })(Form)
    const output = renderer.create(
      <Provider store={store} >
        <ReduxForm />
      </Provider>
    )
    expect(output.toJSON()).toMatchSnapshot()
  })
  it('should render correctly with optional props', () => {
    const input = {},
      meta = {}
    const output = renderer.create(
      <FieldGroup
        controlId='fieldGroupControlId'
        input={input}
        label='a label'
        meta={meta}
        placeholder='a placeholder'
        size='large'
        type='text'
      />
    )
    expect(output.toJSON()).toMatchSnapshot()
  })
})

describe('ValidableFieldGroup', () => {
  it('should render correctly with no error and touched', () => {
    const input = {},
      meta = {
        error: null,
        touched: true
      }
    const output = renderer.create(
      <ValidableFieldGroup
        controlId='fieldGroupControlId'
        input={input}
        meta={meta}
        type='text'
      />
    )
    expect(output.toJSON()).toMatchSnapshot()
  })
  it('should render correctly with error and touched', () => {
    const input = {},
      meta = {
        error: 'some error',
        touched: true
      }
    const output = renderer.create(
      <ValidableFieldGroup
        controlId='fieldGroupControlId'
        input={input}
        meta={meta}
        type='text'
        validationState='error'
      />
    )
    expect(output.toJSON()).toMatchSnapshot()
  })
})
