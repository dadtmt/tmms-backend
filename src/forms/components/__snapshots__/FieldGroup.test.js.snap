exports[`FieldGroup should render correctly password input 1`] = `
<div
  className="form-group form-group-lg">
  <input
    className="form-control"
    id="password"
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    placeholder="password"
    type="password"
    value="" />
  
  <span
    className="help-block" />
</div>
`;

exports[`FieldGroup should render correctly with optional props 1`] = `
<div
  className="form-group form-group-lg">
  <label
    className="control-label"
    htmlFor="fieldGroupControlId">
    a label
  </label>
  <input
    className="form-control"
    id="fieldGroupControlId"
    placeholder="a placeholder"
    type="text" />
  <span
    className="help-block" />
</div>
`;

exports[`FieldGroup should render correctly with required props 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id="fieldGroupControlId"
    placeholder={undefined}
    type="text" />
  <span
    className="help-block" />
</div>
`;

exports[`ValidableFieldGroup should render correctly with error and touched 1`] = `
<div
  className="form-group has-feedback has-error">
  <input
    className="form-control"
    id="fieldGroupControlId"
    placeholder={undefined}
    type="text" />
  <span
    className="form-control-feedback glyphicon glyphicon-remove" />
  <span
    className="help-block">
    some error
  </span>
</div>
`;

exports[`ValidableFieldGroup should render correctly with no error and touched 1`] = `
<div
  className="form-group has-feedback has-success">
  <input
    className="form-control"
    id="fieldGroupControlId"
    placeholder={undefined}
    type="text" />
  <span
    className="form-control-feedback glyphicon glyphicon-ok" />
  <span
    className="help-block" />
</div>
`;
